New chart: Chainloop

javirln opened the pull request:
### Description of the change

This PR adds a new chart to the catalog: [Chainloop](https://github.com/chainloop-dev/chainloop), an open-source software supply chain control plane, the evidence store and a single source of truth for artifacts, metadata plus a declarative attestation crafting process.

This solution is compounded by 4 different components:

- Chainloop controlplane (mandatory)
- Chainloop artifact proxy (mandatory)
- PostgreSQL (optional, enabled by default)
- Vault (optional)

We included support for the following K8s objects for every component:

- Ingress & Ingress TLS secrets.
- HPA & PDB.
- Persistence via PVCs.
- TLS certs for mTLS communications.
- Custom configuration provided via values or existing ConfigMap.
- Network Policies.
- ServiceMonitor for GKE monitoring Operator.

A more in depth explanation and guides to the Chainloop chart can be found [here](https://github.com/chainloop-dev/chainloop/tree/main/deployment/chainloop).

### Additional information

The Helm Chart can be deployed in two different modes, `standard` and `development`.


If `development` is active, a Vault Chart will be installed along with the rest of services for secret management.

![Screenshot 2024-06-11 at 17 02 33](https://github.com/bitnami/charts/assets/4128269/018faedc-f61c-46f7-9e34-92780a08f30f)

While the default deployment mode, `standard` relies on external dependencies to be available in advance.

![Screenshot 2024-06-11 at 17 02 42](https://github.com/bitnami/charts/assets/4128269/41712b2c-d1cd-4c8f-ab09-8464969f1754)

### Important notice

The current implementation of the Chainloop Chart requires a Dex instance configured as an OIDC provider. We found an external Dex chart not provided by Bitnami and considered using it but we were wondering if you would consider to package it as part of your catalog?. Additionally, we noticed that Bitnami's Argo-CD includes Dex templates within its chart. Could this be an option for us as well?

Additionally we keep working on adding more integration tests to the Chart.

### How do we build the images?
The images are built using a GoReleaser action that takes as source of truth a `Dockerfile`. Here the links:
- Controlplane: https://github.com/chainloop-dev/chainloop/blob/main/app/controlplane/Dockerfile.goreleaser
- CAS: https://github.com/chainloop-dev/chainloop/blob/main/app/artifact-cas/Dockerfile.goreleaser
- CLI: https://github.com/chainloop-dev/chainloop/blob/main/app/cli/Dockerfile.goreleaser

All those images are then pushed to GHCR: https://github.com/orgs/chainloop-dev/packages?repo_name=chainloop

All of them based on `scratch`

### Checklist

- [x] Chart version bumped in Chart.yaml according to [semver](http://semver.org/). This is not necessary when the changes - only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm]-(https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

javsalgar commented:
Hi!

Thank you so much for the PR! Could you fix the action errors? Specially the license headers, as we cannot accept a contribution unless the copyright headers match the requirements.

javirln commented:
> Hi!
> 
> Thank you so much for the PR! Could you fix the action errors? Specially the license headers, as we cannot accept a contribution unless the copyright headers match the requirements.

Thanks Salme for the quick reply :)

We have just question regarding the license header changes. We have indeed added them but probably with the wrong scope.

```
diff --git a/bitnami/chainloop/Chart.yaml b/bitnami/chainloop/Chart.yaml
index 291efc4f94..b88420c8f3 100644
--- a/bitnami/chainloop/Chart.yaml
+++ b/bitnami/chainloop/Chart.yaml
@@ -1,4 +1,4 @@
-# Copyright Chainloop, Inc. All Rights Reserved.
+# Copyright Broadcom, Inc. All Rights Reserved.
 # SPDX-License-Identifier: APACHE-2.0
```

What entitles the change on the header? That you folks take ownership and control on Chart from the moment is merged?

migmartri commented:
Hi team, I've made some changes which include

- Use of Bitnami's image structure to make it compatible with Helm dt
- Global options support, for now registry and pull secrets

migmartri commented:
> The current implementation of the Chainloop Chart requires a Dex instance configured as an OIDC provider. We found an external Dex chart not provided by Bitnami and considered using it but we were wondering if you would consider to package it as part of your catalog?. Additionally, we noticed that Bitnami's Argo-CD includes Dex templates within its chart. Could this be an option for us as well?

We will replace upstream dex for a variation of the templates found [here](https://github.com/bitnami/charts/tree/main/bitnami/argo-cd/templates/dex). @javsalgar let me know if you'd prefer if we follow some other approach. To us, we just need an OIDC provider, it doesn't need to be dex, so happy for recommendations if you have any other in the catalog.

Thanks

migruiz4 commented:
Hi @javirln @migmartri,

Thank you very much for your contribution! 

I have noticed that most of the files in this PR haven't been created using the [template chart](https://github.com/bitnami/charts/tree/main/template/CHART_NAME) as base, and therefore missing many features that are common between all the charts in our catalog. 

Some examples:
- The `Deployment` manifests are missing several features under `.spec.template.spec`, including features such as `extraEnvVars`, `extraInitContainers`, `sidecars`, etc.
- The SQL proxy deployment is mostly hardcoded and does not allow any templating.
- The NOTES.txt is missing bitnami checks and should be aligned with the style used in other charts.
- The `Service` manifests are missing features under `.spec`.

I would appreciate it if you could adapt those manifests and include the missing values.

Additionally, I have noticed the gce-proxy image is hardcoded and does not allow users to set it in the `values.yaml`.

I'm currently working on releasing all the required images as part of the Bitnami catalog, including the gce-proxy image. I will let you know once they are available so we can update the chart to use them.

migmartri commented:
> Hi @javirln @migmartri,
> 
> Thank you very much for your contribution!
> 
> I have noticed that most of the files in this PR haven't been created using the [template chart](https://github.com/bitnami/charts/tree/main/template/CHART_NAME) as base, and therefore missing many features that are common between all the charts in our catalog.
> 
> Some examples:
> 
> * The `Deployment` manifests are missing several features under `.spec.template.spec`, including features such as `extraEnvVars`, `extraInitContainers`, `sidecars`, etc.
> * The SQL proxy deployment is mostly hardcoded and does not allow any templating.
> * The NOTES.txt is missing bitnami checks and should be aligned with the style used in other charts.
> * The `Service` manifests are missing features under `.spec`.
> 
> I would appreciate it if you could adapt those manifests and include the missing values.
> 
> Additionally, I have noticed the gce-proxy image is hardcoded and does not allow users to set it in the `values.yaml`.
> 
> I'm currently working on releasing all the required images as part of the Bitnami catalog, including the gce-proxy image. I will let you know once they are available so we can update the chart to use them.

Thanks @migruiz4, we'll work on those

migmartri commented:
> I'm currently working on releasing all the required images as part of the Bitnami catalog, including the gce-proxy image. I will let you know once they are available so we can update the chart to use them.

Hi @migruiz4. Please do not package the `gce-proxy` image. We've decided to remove it from our Chart. Thanks

javirln commented:
Hello @migruiz4 :)

I have a problem with the following check: https://github.com/bitnami/charts/actions/runs/10247365041/job/28346395566?pr=27100

The CI is detecting I have changed 2 charts rather than 1. It's somehow true since what I have changed is the main Chainloop chart and one sub chart of it, `Dex` and it's detecting those are two different charts since both of them are within `bitnami/` folder.

Is there I a way to make it pass?

migruiz4 commented:
Hi @javirln,

I'm sorry but we do not support having subcharts defined at `bitnami/<appname>/charts/<subchart>`. Subcharts need to be defined at Chart.yaml using an OCI reference.

In other words, subcharts must exist as charts themselves.

migruiz4 commented:
In this case, I would suggest handling `dex` as an additional component of the chart, sharing the same base `values.yaml`, same way we do for `argo-cd`:

- Add `templates/dex` folder.
- Use .Values.dex at `values.yaml`.

An alternative would be to create a separate PR with another new chart `bitnami/dex`, but I would need to check it with the team first.

migmartri commented:
Hi @migruiz4, do all the sub-charts also need to be Bitnami charts (we could package this chart and push it to OCI; note that it uses Bitnami images)? I'd rather not pollute our main chart configuration with the whole dependency `values.yaml`. Dex is a dependency, specifically for development mode, and ideally, it should be treated as such, not as a main component.

If that's not an option, as last resort then I'd do what you suggested, thanks!

migmartri commented:
> Hi @migruiz4, do all the sub-charts also need to be Bitnami charts (we could package this chart and push it to OCI; note that it uses Bitnami images)? I'd rather not pollute our main chart configuration with the whole dependency `values.yaml`'s Dex as a dependency, specifically for development mode, and ideally, it should be treated as such, not as a main component.
> 
> If that's not an option, as last resort then I'd do what you suggested, thanks!

Hi @migruiz4, I just checked with @javirln, and we are going to move the templates to the parent chart for the Bitnami case; stay tuned. Thanks for the guidance :)

javirln commented:
> In this case, I would suggest handling `dex` as an additional component of the chart, sharing the same base `values.yaml`, same way we do for `argo-cd`:
> 
> * Add `templates/dex` folder.
> * Use .Values.dex at `values.yaml`.
> 
> An alternative would be to create a separate PR with another new chart `bitnami/dex`, but I would need to check it with the team first.

Thanks @migruiz4, all changes applied 😄

migruiz4 commented on bitnami/chainloop/templates/dex/secret.yaml:
```diff
@@ -0,0 +1,44 @@
+{{- /*
```
Same feedback about the hardcoded ports here

migruiz4 commented on bitnami/chainloop/templates/controlplane/deployment.yaml:
```diff
@@ -0,0 +1,217 @@
+                  name: {{ include "chainloop.controlplane.fullname" . }}
+                  key: db_migrate_source
+      containers:
+        - name: {{ .Chart.Name }}
```
We usually hardcode a container name, in this case we could use 'controlplane'

migruiz4 commented on bitnami/chainloop/templates/cas/deployment.yaml:
```diff
@@ -0,0 +1,175 @@
+          {{- include "common.tplvalues.render" (dict "value" .Values.cas.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
```
Same comment here, we usually hardcode a container name, in this case we could use 'cas'.

migruiz4 commented on bitnami/chainloop/templates/dex/deployment.yaml:
```diff
@@ -0,0 +1,195 @@
+      shareProcessNamespace: {{ .Values.dex.shareProcessNamespace }}
+      {{- end }}
+      {{- if .Values.dex.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.controller.topologySpreadConstraints "context" .) | nindent 8 }}
```
```suggestion
      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.dex.topologySpreadConstraints "context" .) | nindent 8 }}
```

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2046 @@
+## @param vault.server.extraEnvVars[1].name Address to listen on development mode
+## @param vault.server.extraEnvVars[1].value The address to listen on. Default: [::]:8200
+vault:
+  server:
+    args: [
+      "server",
+      "-dev"
+    ]
+    extraEnvVars:
+      - name: VAULT_DEV_ROOT_TOKEN_ID
+        value: "notasecret"
+      - name: VAULT_DEV_LISTEN_ADDRESS
+        value: "[::]:8200"
+    config: "storage \"inmem\" {}\ndisable_mlock = true\nui = true\nservice_registration \"kubernetes\" {}"
```
I coudn't find any references to `vault.server.args` or `vault.server.config` in the templates. Are these values used anywhere?

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2046 @@
+    ## @param controlplane.autoscaling.vpa.maxAllowed VPA Max allowed resources for the pod
+    ## @param controlplane.autoscaling.vpa.minAllowed VPA Min allowed resources for the pod
+    ##
+    vpa:
```
VPA values defined  but missing VPA template.

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2046 @@
+  ## @param cas.pdb.minAvailable Minimum number/percentage of pods that should remain scheduled
+  ## @param cas.pdb.maxUnavailable Maximum number/percentage of pods that may be made unavailable. Defaults to `1` if both `cas.pdb.minAvailable` and `cas.pdb.maxUnavailable` are empty.
+  ##
+  pdb:
```
PDB defined but missing PDB templates

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2046 @@
+    ##
+    sessionAffinityConfig: {}
+
+  serviceAPI:
```
`https` values are defined but never used

migruiz4 reviewed (commented):
Thank you very much for adapting the chart to the bitnami standards!
Please take a look at my comments.

Additionally, I have noticed that the chart does not include networkPolicy files

javirln commented:
Thanks @migruiz4 for the review.

Let me go little by little fixing things and I'll mark the PR as ready to review again.

migruiz4 commented:
No problem! Let me know if there is anything I can do to help!

As mentioned in the review comments, the Bitnami images for chainloop components are now available:
   - https://hub.docker.com/r/bitnami/chainloop-control-plane
   - https://hub.docker.com/r/bitnami/chainloop-control-plane-migrations
   - https://hub.docker.com/r/bitnami/chainloop-artifact-cas

Let me know when the chart is ready for review again and we can enable the 'verify' label to start testing the chart with the Github CI pipeline.

Some additional notes about the review:
- There were some missing manifests (VPA, PDB and NetworkPolicy) that you can find in the template chart and other charts in the catalog.
- I would recommend reducing the usage of `@extra` and `@skip` in the values.yaml 
- Having the possibility of providing existing secrets and configmaps would be a 'Good to have' feature, but I would not block the release if it becomes too painful to implement.

javirln reviewed (commented):
First round of feedback addressed. Haven't gone through the README, I'll submit another commit for it.

In the meantime, I haven't change the images since they don't seem to work just by replacing them.

javirln commented:
@migruiz4 I've tested both docker images for the artifact-cas, upstream and bitnami one and they differ in the output:

Upstream
```shell
$ docker run --rm ghcr.io/chainloop-dev/chainloop/artifact-cas:latest
Unable to find image 'ghcr.io/chainloop-dev/chainloop/artifact-cas:latest' locally
latest: Pulling from chainloop-dev/chainloop/artifact-cas
Digest: sha256:74a774ccda139160162d2c522b8212e51cecc0d74194c9851cb82c81595c5371
Status: Downloaded newer image for ghcr.io/chainloop-dev/chainloop/artifact-cas:latest
2024/08/07 11:39:42 maxprocs: Leaving GOMAXPROCS=11: CPU quota undefined
panic: stat /data/conf: no such file or directory

goroutine 1 [running]:
main.main()
	/home/runner/work/chainloop/chainloop/app/artifact-cas/cmd/main.go:98 +0x637
```

Bitnami
```shell
$ docker run --rm --name chainloop-artifact-cas bitnami/chainloop-artifact-cas:latest
2024/08/07 11:36:13 maxprocs: Leaving GOMAXPROCS=11: CPU quota undefined
panic: stat ../../configs: no such file or directory

goroutine 1 [running]:
main.main()
	/bitnami/blacksmith-sandox/chainloop-artifact-cas-0.95.3/src/github.com/chainloop-dev/chainloop/app/artifact-cas/app/artifact-cas/cmd/main.go:98 +0x568
```

Is that image assuming a specific config directory?

javirln commented:
> @migruiz4 I've tested both docker images for the artifact-cas, upstream and bitnami one and they differ in the output:
> 
> Upstream
> 
> ```shell
> $ docker run --rm ghcr.io/chainloop-dev/chainloop/artifact-cas:latest
> Unable to find image 'ghcr.io/chainloop-dev/chainloop/artifact-cas:latest' locally
> latest: Pulling from chainloop-dev/chainloop/artifact-cas
> Digest: sha256:74a774ccda139160162d2c522b8212e51cecc0d74194c9851cb82c81595c5371
> Status: Downloaded newer image for ghcr.io/chainloop-dev/chainloop/artifact-cas:latest
> 2024/08/07 11:39:42 maxprocs: Leaving GOMAXPROCS=11: CPU quota undefined
> panic: stat /data/conf: no such file or directory
> 
> goroutine 1 [running]:
> main.main()
> 	/home/runner/work/chainloop/chainloop/app/artifact-cas/cmd/main.go:98 +0x637
> ```
> 
> Bitnami
> 
> ```shell
> $ docker run --rm --name chainloop-artifact-cas bitnami/chainloop-artifact-cas:latest
> 2024/08/07 11:36:13 maxprocs: Leaving GOMAXPROCS=11: CPU quota undefined
> panic: stat ../../configs: no such file or directory
> 
> goroutine 1 [running]:
> main.main()
> 	/bitnami/blacksmith-sandox/chainloop-artifact-cas-0.95.3/src/github.com/chainloop-dev/chainloop/app/artifact-cas/app/artifact-cas/cmd/main.go:98 +0x568
> ```
> 
> Is that image assuming a specific config directory?
> 
> The upstream one of controlplane and artifact-cas have as entrypoint:
> 
> ```
> ENTRYPOINT [ "./control-plane", "--conf", "/data/conf"]
> ENTRYPOINT [ "./artifact-cas", "--conf", "/data/conf"]
> ```
> 
> I'll try to work on the change.

Forget about this, I've made them work :D

javirln commented:
> No problem! Let me know if there is anything I can do to help!
> 
> As mentioned in the review comments, the Bitnami images for chainloop components are now available:
> 
> * https://hub.docker.com/r/bitnami/chainloop-control-plane
> * https://hub.docker.com/r/bitnami/chainloop-control-plane-migrations
> * https://hub.docker.com/r/bitnami/chainloop-artifact-cas
> 
> Let me know when the chart is ready for review again and we can enable the 'verify' label to start testing the chart with the Github CI pipeline.
> 
> Some additional notes about the review:
> 
> * There were some missing manifests (VPA, PDB and NetworkPolicy) that you can find in the template chart and other charts in the catalog.
> * I would recommend reducing the usage of `@extra` and `@skip` in the values.yaml
> * Having the possibility of providing existing secrets and configmaps would be a 'Good to have' feature, but I would not block the release if it becomes too painful to implement.

Thanks @migruiz4! 

I've replaced the images for the bitnami ones and tested with a few tweaks that they are working. Aside from the questions I've raised on your comments a few more:
- Are the `VPA, PDB and NetworkPolicy` required to have? Do we have to include them?
- I've tried to reduce the `@extra` and `@skip` but in the most cases those are optional parameters to configure the chart.

I would like to verify the `ginkgo` tests are also working.

migruiz4 commented:
> Are the VPA, PDB and NetworkPolicy required to have? Do we have to include them?

It would be recommended, especially having a Networkpolicy is a security feature that helps increase the overall kubescape score of the chart.

> I've tried to reduce the @extra and @skip but in the most cases those are optional parameters to configure the chart.

While they are optional, the effect would be similar but as mentioned, it has an impact. Let me explain:

Considering below template:
```
{{- if .Values.exampleValue }}
...
{{- end }}
```
Behaves the same way if any of the following is used at values.yaml
```
exampleValue: ""
exampleValue: []
exampleValue: {}
exampleValue: false
```
It also behaves the same way as using this
```
## @skip/@extra exampleValue ...
# exampleValue: ""
```
The main difference is that using the first option provides information about the value schema, and helps us better
document the value and its default in the README.

> I would like to verify the ginkgo tests are also working.

I will now enable testing, thanks!

migruiz4 commented on bitnami/chainloop/Chart.yaml:
```diff
@@ -0,0 +1,66 @@
+    version: 1.4.x
+description: Chainloop is an open source software supply chain control plane, a single source of truth for artifacts plus a declarative attestation crafting process.
+home: https://bitnami.com
+icon: https://bitnami.com/assets/stacks/chainloop-control-plane/img/chainloop-control-plane-stack-220x234.png
```
```suggestion
icon: https://bitnami.com/assets/stacks/chainloop/img/chainloop-stack-220x234.png
```

migruiz4 commented on bitnami/chainloop/Chart.yaml:
```diff
@@ -0,0 +1,38 @@
+# Copyright Broadcom, Inc. All Rights Reserved.
```
Thank you for updating the Chart.yaml!
For the image I sent a new comment with the link

migruiz4 commented on bitnami/chainloop/templates/NOTES.txt:
```diff
@@ -0,0 +1,62 @@
+
+** Please be patient while the chart is being deployed **
+
+###########################################################################
+  DEVELOPMENT MODE
+###########################################################################
+
+██████╗ ███████╗██╗    ██╗ █████╗ ██████╗ ███████╗
+██╔══██╗██╔════╝██║    ██║██╔══██╗██╔══██╗██╔════╝
+██████╔╝█████╗  ██║ █╗ ██║███████║██████╔╝█████╗
+██╔══██╗██╔══╝  ██║███╗██║██╔══██║██╔══██╗██╔══╝
+██████╔╝███████╗╚███╔███╔╝██║  ██║██║  ██║███████╗
+╚═════╝ ╚══════╝ ╚══╝╚══╝ ╚═╝  ╚═╝╚═╝  ╚══════╝
+
+Instance running in development mode!
+
+Development mode, by default
+
+- Runs an insecure, unsealed, non-persistent instance of Vault
+- Is configured with development authentication keys
```
This is missing the conditional `{{- if .Values.development }}` right?

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2254 @@
+  #    - name: ""
+  #      role: ""
+
+  ## @extra controlplane.prometheus_org_metrics List of organizations to expose metrics for using Prometheus
+  #  prometheus_org_metrics:
+  #    - org_name: "read-only-demo"
```
```suggestion
  ## @param controlplane.prometheus_org_metrics List of organizations to expose metrics for using Prometheus
  ##  e.g.:
  ##  prometheus_org_metrics:
  ##    - org_name: "read-only-demo"
  prometheus_org_metrics: []
```

migruiz4 commented on bitnami/chainloop/templates/controlplane/deployment.yaml:
```diff
@@ -0,0 +1,220 @@
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.controlplane.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.controlplane.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.controlplane.podSecurityContext "enabled" | toYaml | nindent 8 }}
```
I'm sorry I know this comes from a typo in the template:
```suggestion
          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.controlplane.podSecurityContext "context" $) | nindent 8 }}
```

migruiz4 commented on bitnami/chainloop/templates/cas/deployment.yaml:
```diff
@@ -0,0 +1,177 @@
+      {{- if .Values.cas.topologySpreadConstraints }}
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.cas.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.cas.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.cas.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
```
```suggestion
      {{- if .Values.cas.podSecurityContext.enabled }}
      securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.cas.podSecurityContext "context" $) | nindent 8 }}
      {{- end }}
```

migruiz4 commented on bitnami/chainloop/templates/cas/deployment.yaml:
```diff
@@ -0,0 +1,177 @@
+        {{- end }}
+      containers:
+        - name: cas
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.cas.containerSecurityContext "context" $) | nindent 12 }}
```
```suggestion
          {{- if .Values.cas.containerSecurityContext.enabled }}
          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.cas.containerSecurityContext "context" $) | nindent 12 }}
          {{- end }}
```

migruiz4 commented on bitnami/chainloop/templates/controlplane/deployment.yaml:
```diff
@@ -0,0 +1,220 @@
+                  key: db_migrate_source
+      containers:
+        - name: controlplane
+          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.controlplane.containerSecurityContext "context" $) | nindent 12 }}
```
```suggestion
          {{- if .Values.controlplane.containerSecurityContext.enabled }}
          securityContext: {{- include "common.compatibility.renderSecurityContext" (dict "secContext" .Values.controlplane.containerSecurityContext "context" $) | nindent 12 }}
          {{- end }}

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2254 @@
+  ## Network Policies
+  ## Ref: https://kubernetes.io/docs/concepts/services-networking/network-policies/
+  ##
+  networkPolicy:
```
Dex NetworkPolicy is missing

migruiz4 commented on bitnami/chainloop/values.yaml:
```diff
@@ -0,0 +1,2254 @@
+    ##
+    ports:
+      http: 80
+      https: 443
```
I think you missed this from the previous review.
Please remove the `https` values since they are not being used.

migruiz4 commented on bitnami/chainloop/Chart.yaml:
```diff
@@ -0,0 +1,66 @@
+      name: dex
+apiVersion: v2
+appVersion: 0.95.3
+dependencies:
+  - name: common
+    repository: https://charts.bitnami.com/bitnami
+    tags:
+      - bitnami-common
+    version: 2.x.x
+  - condition: postgresql.enabled
+    name: postgresql
+    repository: https://charts.bitnami.com/bitnami
+    version: 15.x.x
+  - condition: development
+    name: vault
+    repository: https://charts.bitnami.com/bitnami
+    version: 1.4.x
```
Subcharts must use the bitnami OCI registry
```suggestion
dependencies:
  - name: common
    repository: oci://registry-1.docker.io/bitnamicharts
    tags:
      - bitnami-common
    version: 2.x.x
  - condition: postgresql.enabled
    name: postgresql
    repository: oci://registry-1.docker.io/bitnamicharts
    version: 15.x.x
  - condition: development
    name: vault
    repository: oci://registry-1.docker.io/bitnamicharts
    version: 1.4.x
```

migruiz4 reviewed (commented):
Thank you very much for implementing the feedback!
Some additional comments, in this case they are mostly minor details

javirln commented:
@migruiz4 everything addressed, thanks again for the review :D

migmartri commented:
Ditto, thanks @migruiz4 for the detailed review :)

On Thu, Aug 8, 2024, 6:08 PM Javier Rodríguez ***@***.***>
wrote:

> @migruiz4 <https://github.com/migruiz4> everything addressed, thanks
> again for the review :D
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bitnami/charts/pull/27100#issuecomment-2276186352>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAF7S5C6YRHHYARJHAJTLLZQOJXFAVCNFSM6AAAAABJEQZIS6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENZWGE4DMMZVGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

migruiz4 reviewed (approved):
Amazing work @javirln @migmartri, thank you very much!

migmartri commented:
Thanks a lot, @migruiz4, for the thorough review; you helped us a lot to improve our chart 🥇 

Re: the [image used](https://bitnami.com/assets/stacks/chainloop/img/chainloop-stack-220x234.png) in the chart and Bitnami listing, could you use this logo instead?

![loop black on white gradient](https://github.com/user-attachments/assets/a584bf84-5de0-49ec-937d-445f15449bd4)

or this one?

![loop white on black gradient](https://github.com/user-attachments/assets/65f4ebd6-b1a4-4647-a2db-9386214ed94f)

Thanks a lot!

javsalgar commented:
Hi!

You mean without the bitnami hexagon?

migmartri commented:
> Hi!
> 
> You mean without the bitnami hexagon?

No, the hexagon is fine; it's just that's the one you put; it's a slightly different infinite loop icon.

The icon you used

![image](https://github.com/user-attachments/assets/87d737bb-633f-4fd7-b3ad-ea2cfbd8fc00)

The icon I am proposing to put in the hexagon

![image](https://github.com/user-attachments/assets/5fe84816-f715-478d-870c-13b4d55d6eac)

Thanks

javsalgar commented:
Hi! I used the image and it shows like this. Does it work? Or maybe the logo is too small? 

![chainloop-stack-220x234](https://github.com/user-attachments/assets/f2b9841e-fa3c-40f7-bece-bf46da1f4972)

migmartri commented:
hi @javsalgar, this looks ok, but it would be great, if possible, to try this version, which removed most of the padding around the logo.

Thanks for checking on this @javsalgar, you guys rock :)

![rect295](https://github.com/user-attachments/assets/d5df7df1-e1e6-4f63-879b-07fcc3df171c)

javsalgar commented:
Could you remove the black border? It's doing something strange with the logo

migmartri commented:
hi @javsalgar, sorry I didn't understand the request, what black border?

javsalgar commented:
Never mind, I fixed it myself. This is how it looks now

![chainloop-gcp-130x130](https://github.com/user-attachments/assets/bfb6c9aa-0263-4b8c-aad4-dbe6aa01bb3d)
